src/main/scala/it/pagopa/interop/attributeregistryprocess/api/impl/AttributeRegistryApiServiceImpl.scala

```diff
@@ -101,4 +107,80 @@ final case class AttributeRegistryApiServiceImpl(
       }
     }
   }
+
+  override def loadCertifiedAttributes()(implicit contexts: Seq[(String, String)]): Route =
+    authorize(INTERNAL_ROLE) {
+      val operationLabel: String = s"Loading certified attributes from Party Registry"
+      logger.info(operationLabel)
+
+      val result: Future[Unit] = for {
+        bearer     <- getFutureBearer(contexts)
+        categories <- partyRegistryService.getCategories(bearer)
+        attributeSeedsCategories   = categories.items.map(c =>
+          AttributeSeed(
+            code = Option(c.code),
+            kind = AttributeKind.CERTIFIED,
+            description = c.name, // passing the name since no description exists at party-registry-proxy
+            origin = Option(c.origin),
+            name = c.name
+          )
+        )
+        institutions <- partyRegistryService.getInstitutions(bearer)
+        attributeSeedsInstitutions = institutions.items.map(i =>
+          AttributeSeed(
+            code = Option(i.originId),
+            kind = AttributeKind.CERTIFIED,
+            description = i.description,
+            origin = Option(i.origin),
+            name = i.description
+          )
+        )
+
+        _ <- addNewAttributes(attributeSeedsCategories ++ attributeSeedsInstitutions)
+      } yield ()
+
+      onComplete(result) {
+        loadCertifiedAttributesResponse[Unit](operationLabel)(_ => loadCertifiedAttributes200)
+      }
+    }
+
+  def addNewAttributes(
+    attributeSeed: Seq[AttributeSeed]
+  )(implicit contexts: Seq[(String, String)]): Future[Set[Attribute]] = {
+
+    case class DeltaAttributes(attributes: Set[Attribute], seeds: Set[AttributeSeed]) {
+      def addAttribute(attr: Attribute): DeltaAttributes = copy(attributes = attributes + attr)
+
+      def addSeed(seed: AttributeSeed): DeltaAttributes = copy(seeds = seeds + seed)
+    }
+
+    // calculating the delta of attributes
+    def delta(attrs: List[Attribute]): DeltaAttributes =
+      attributeSeed.foldLeft[DeltaAttributes](DeltaAttributes(Set.empty, Set.empty))((delta, seed) =>
+        attrs
+          .find(persisted => seed.name.equalsIgnoreCase(persisted.name))
+          .fold(delta.addSeed(seed))(delta.addAttribute)
+      )
+
+    // for all the not existing attributes, execute the command to persist them through event sourcing
+    for {
+      attributesfromRM <- getAll(50)(readModelService.find[PersistentAttribute]("attributes", Filters.empty(), _, _))
```

Comment:
instead of retrieving all attributes and work on them, a different approach could be to request a batch (like 50), process them and then retrieve the next batch.
This way the resources consumption (mostly the memory) will be constant despite the number of attributes in the registry, instead of growing proportionally